src/Type/StringType.php

```diff
@@ -137,6 +137,11 @@ public function toArray(): Type
 		);
 	}
 
+	public function toArrayKey(): Type
+	{
+		return $this;
```

Comment:
This needs to return most likely `int|string` I believe?

Reply:
It'd be correct but annoying.

Reply:
It didn't do so in the previous implementation, and though technically that is correct I don't think changing it now is wise.

Reply:
ah right, I thought the failing tests were related, but looks like they aren't :)